src/match.js

```diff
@@ -101,6 +111,20 @@ function matchMultiple(expr, str, match_length, min_length = 0, max_length = -1)
   return [false, 0];
 }
 
+function matchAlternate(expr, str, match_length) {
+  const [head, operator, rest] = splitExpr(expr);
+  const options = splitAlternate(head);
+  for (const option of options) {
+    const [matched, new_match_length] = matchExpr(
+      option + rest, 
+      str,
+      match_length
+    );
+    if (matched) return [matched, new_match_length];
```

Comment:
It would be better to name all values in camel case. And since 'matched' have type bool it would be clearer to call it 'isMatched'
```suggestion
    const [isMatched, newMatchLength] = matchExpr(
      option + rest, 
      str,
      matchLength
    );
    if (isMatched) return [isMatched, newMatchLength];
```

Reply:
The code will be refactored